foundryContract/src/BatchSubmission.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./interfaces/IAggregatorSubmit.sol";
+
+contract BatchSubmission is Ownable {
+    uint256 maxSubmission = 50;
+    address[] public oracleAddresses;
+    mapping(address => bool) oracles;
+
+    error OnlyOracle();
+    error InvalidOracle();
+    error InvalidSubmissionLength();
+
+    event OracleAdded(address oracle);
+    event OracleRemoved(address oracle);
+    event MaxSubmissionSet(uint256 maxSubmission);
+
+    constructor() Ownable(msg.sender) {}
+
+    function getOracles() public view returns (address[] memory) {
+        return oracleAddresses;
+    }
+
+    function addOracle(address _oracle) public onlyOwner {
+        oracleAddresses.push(_oracle);
+        oracles[_oracle] = true;
+        emit OracleAdded(_oracle);
+    }
+
+    function removeOracle(address _oracle) public onlyOwner {
```

Comment:
I think the structure of this function could be a bit better, users won't suffer because it is only for admin, but I think the common way to implement it might be as follows.

```Solidity
function removeOracle(address _oracle) public onlyOwner {
      if (!oracles[_oracle]) {
          emit InvalidOracle();
          return;
      }

      for (uint256 i = 0; i < oracleAddresses.length; ++i) {
          if (oracleAddresses[i] == _oracle) {
              address last = oracleAddresses[oracleAddresses.length - 1];
              oracleAddresses[i] = last;
              oracleAddresses.pop();
              delete oracles[_oracle];
              emit OracleRemoved(_oracle);
              break;
          }
      }
  }
```